inst/include/dust/utils.hpp

```diff
@@ -64,6 +64,37 @@ HOSTDEVICE T epsilon() {
 #endif
 }
 
+__nv_exec_check_disable__
+inline int integer_max() {
+#ifdef __CUDA_ARCH__
+  return INT_MAX;
+#else
+  return std::numeric_limits<int>::max();
+#endif
+}
+
+template <typename T>
+T lgamma(T x);
+
+// We need this for the lgamma in rpois to work
+#ifdef __CUDA_ARCH__
```

Comment:
I don't think this ifdef is needed, you should just be able to mark the functions with DEVICE and HOST (which you have already done – except HOST is implicit, but I think it helps to make it explicit in this case)

Reply:
Also I think this ifdef will never be true